File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int n = progresses.size();
    vector<int> days_left(n);

    //배포 예상일 7, 3, 9
    for(int i=0; i < n; i++) days_left[i] = ceil((100.0-progresses[i])/speeds[i]);
    
    int count = 0;
    int max_day = days_left[0];
    
    for(int i=0; i<n; i++){
        if(days_left[i] <= max_day) count ++;
        else{
            answer.push_back(count);
            max_day = days_left[i];
            count = 1;
        }
    }
    
    answer.push_back(count);
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    # 처음 거 기준으로 배포할 때 다음 값들도 배포되는지 확인
    
    progresses = deque(progresses)
    speeds = deque(speeds)
    
    answer = []
    while progresses:
        job = progresses[0]
        days = math.ceil((100 - job) / speeds[0])

        cnt = 1
        for i in range(1, len(progresses)):
            if progresses[i] + speeds[i] * days >= 100:
                cnt += 1
            else:
                break
        
        answer.append(cnt)

        for i in range(cnt):
            progresses.popleft()
            speeds.popleft()
        
    return answer